Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm

```diff
@@ -128,15 +128,43 @@ explicit RemoteLayerTreeDisplayLinkClient(WebPageProxyIdentifier pageID)
     observerID = { };
 }
 
+void RemoteLayerTreeDrawingAreaProxyMac::layoutBannerLayers(const RemoteLayerTreeTransaction& transaction)
+{
+    float totalContentHeight = transaction.contentsSize().height();
+
+    auto layoutBannerLayer = [](CALayer *bannerLayer, float y, float width) {
+        if (bannerLayer.frame.origin.y == y && bannerLayer.frame.size.width == width)
+            return;
+        [CATransaction begin];
+        [CATransaction setAnimationDuration:0];
+        [CATransaction setDisableActions:YES];
+        [bannerLayer setFrame:CGRectMake(0, y, width, bannerLayer.frame.size.height)];
+        [CATransaction commit];
+    };
+
+    if (auto* headerBannerLayer = m_webPageProxy.headerBannerLayer()) {
+        totalContentHeight += headerBannerLayer.frame.size.height;
+        layoutBannerLayer(headerBannerLayer, 0, size().width());
+    }
+
+    if (auto* footerBannerLayer = m_webPageProxy.footerBannerLayer()) {
+        totalContentHeight += footerBannerLayer.frame.size.height;
```

Comment:
Isn't this moving it too far down by the footer height?

Reply:
Yes this is right. I had to use `FrameView::yPositionForHeaderLayer() `and `FrameView::yPositionForFooterLayer()` to calculate the right y position when scrolling the window.